community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/UpgradableDatabase.java

```diff
@@ -84,6 +87,13 @@ public RecordFormats checkUpgradeable( File storeDirectory )
             return format;
         }
 
+        // If we are trying to open an enterprise store when configured to use community format, then inform the user
+        // of the config setting to change since downgrades aren't possible but the store can still be opened.
+        if ( result.actualVersion.startsWith( "vE" ) && !format.storeVersion().startsWith( "vE" ) )
```

Comment:
I think we should not check version strings like this. It is better to check upgradability somehow by looking at `fromFormat` and `format`. Maybe even add a method like `RecordFormats#canBeUpgradesFrom(RecordFormats)` and then create `Result` with outcome `unexpectedUpgradingStoreFormat`. Implementation of `#canBeUpgradesFrom()` can simply check string equality for now.